.github/workflows/reports.yml

```diff
@@ -54,16 +55,108 @@ jobs:
                     }
                   ]
                 }
+              ] +
+              (. | [_nwise(5)] | map({
+                "type": "section",
+                "fields": (. | map([
+                  {
+                    type: "plain_text",
+                    text: .[0]
+                  },
+                  {
+                    type: "mrkdwn",
+                    text: (.[1] + " (<https://portal.azure.com/#blade/AppInsightsExtension/DetailsV2Blade/DataModel/%7B%22eventId%22:%22" + .[2] + "%22,%22timestamp%22:%22" + .[3] + "%22%7D/ComponentId/%7B%22Name%22:%22nomisapi-prod%22,%22ResourceGroup%22:%22nomisapi-prod-rg%22,%22SubscriptionId%22:%22${{ secrets.APP_INSIGHTS_SUBSCRIPTION_ID }}%22%7D|Details>).")
+                  }
+                ]) | flatten)
+              })) +
+              [
+                {
+                  "type": "context",
+                  "elements": [
+                    {
+                      "type": "mrkdwn",
+                      "text": ">This report was generated automatically. For more information, contact the probation integration team."
+                    }
+                  ]
+                }
+              ])
+            }')" | tee -a "$GITHUB_OUTPUT"
+        env:
+          search_result: ${{ steps.search.outputs.result }}
+
+      - name: Send message to Slack
+        id: send
+        if: steps.transform.outputs.result != null
+        uses: slackapi/slack-github-action@e28cf165c92ffef168d23c5c9000cffc8a25e117 # v1.24.0
+        with:
+          channel-id: topic-pi-workforce-allocation
+          payload: ${{ steps.transform.outputs.result }}
+        env:
+          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+
+  approved-premises-failures:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Search app insights
+        id: search
+        run: |
+          echo "result=$(curl -fsSL -H "x-api-key: $key" --data-urlencode "query=$query" --get "$url")" | tee -a "$GITHUB_OUTPUT"
+        env:
+          url: https://api.applicationinsights.io/v1/apps/${{ secrets.APP_INSIGHTS_APPLICATION_GUID }}/query
+          key: ${{ secrets.APP_INSIGHTS_API_KEY }}
+          query: |
+            customEvents
+            | where timestamp > ago(7d) and timestamp < now()
+            | where cloud_RoleName in ("approved-premises-and-delius")
+            | where name in ("ApprovedPremisesFailureReport")
+            | project customDimensions.crn, customDimensions.reason, customDimensions.externalReference, itemId, timestamp
```

Comment:
Do you need the externalReference here?  Or are you planning to include it in the table?